sklearn/pipeline.py

```diff
@@ -132,6 +132,19 @@ def __init__(self, steps, memory=None, verbose=False):
         self.verbose = verbose
         self._validate_steps()
 
+    @property
+    def _warmstartable_parameters(self):
+        # This property exposes the _warmstartable_parameters attribute, e.g.
+        # ['+last_step_name__param']
+        # We consider that only the last step can be warm-started. The first
```

Comment:
Do we need to make this design choice? I may have asked this question before, but don't remember your logic behind it.

Reply:
A pipeline is either:

- a sequence of transformers
- a sequence of transformers + a predictor as the last step

It's safe to assume that transformers are not warm-startable. Maybe in the future one of them will be?? We can worry about that when that happens.

Reply:
I agree we should leave the possibility open, and make the code future-proof. This is one of the reasons why `_warmstartable_parameters` is a list.

But I don't think we should implement support for that, we have no use-case ATM.

Concretely, supporting warm-start for transformers right now is writing code that isn't used (that would require updating the `_fit` method that fits all the transformers)

Reply:
I see. Yeah fair.